clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -850,6 +1034,13 @@ CodeGenFunction::emitBuiltinObjectSize(const Expr *E, unsigned Type,
     }
   }
 
+  if (IsDynamic) {
+    // Emit special code for a flexible array member with the "counted_by"
+    // attribute.
+    if (Value *V = emitFlexibleArrayMemberSize(E, Type, ResType))
+      return V;
```

Comment:
Should `emitFlexibleArrayMemberSize` be made infallible (via asserts rather than early returns of `nullptr`)?